src/main/java/com/varukha/onlinebookstore/service/user/impl/UserServiceImpl.java

```diff
@@ -54,4 +62,17 @@ private Role getDefaultRole() {
                 .orElseThrow(() -> new NoSuchElementException("The role "
                         + ROLE_USER + " is not found in the database"));
     }
+
+    @Override
+    public User getAuthenticatedUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        return userRepository.findUserByEmail(authentication.getName()).orElseThrow(
+                () -> new UsernameNotFoundException("Can't find a user by provided email "
+                        + authentication.getName())
+        );
+    }
+
+    private void createShoppingCart(User user) {
+        shoppingCartService.create(user);
+    }
```

Comment:
Where do you use it?

Reply:
In register method.

Reply:
Why do you need an authenticated user in register method?